src/test/java/com/training/services/tests/ProductsTest.java

```diff
@@ -62,8 +66,88 @@ public void testShorexProductTypeValidate() {
             softAssert.assertThat(
                     response.getPayload().getProducts().get(i).getProductType().getProductType()
                             .equalsIgnoreCase("SHOREX"))
-                    .as("Product id " + response.getPayload().getProducts().get(i).getProductID() +" is not SHOREX")
+                    .as("Product id " + response.getPayload().getProducts().get(i).getProductID() + " is not SHOREX")
                     .isTrue();
         }
     }
+
+    /**
+     * Checking if offeringDate, day  should be less than or equal to 10 days, from sailing date 20200809
+     *
+     * @throws ParseException
+     */
+    @Test
+    public void testShorexOfferingDateValidate() throws ParseException {
+        response = new RestEngine().getResponseGet(baseURL, headerMap, queryParam)
+                .as(Response.class);
+
+        Map<String, Integer> day_Month_daysInMonthFromLink = getDay_Month_daysInMonth(map.get("sailingIdParamValue").split("AL")[1]);
+
+        for (int index = 0; index < response.getPayload().getProducts().size(); index++) {
+            for (int getDateOffering = 0; getDateOffering < response.getPayload().getProducts().get(index).getOffering().size(); getDateOffering++) {
+
+                Map<String, Integer> day_Month_daysInMonth = getDay_Month_daysInMonth(response.getPayload().getProducts().get(index).getOffering().get(getDateOffering).getOfferingDate());
```

Comment:
lets create a variable : 
`List <Offering> offerings = products.get(index).getOffering();`
and use it here and it can reused later in the code as well.
this will enhance readability